### PR TITLE
Add jBallerina debugger integration tests

### PR DESCRIPTION
**Pull request by praveennadarajah**
## Purpose
This PR will add jBallerina debugger integration tests

### CODE REVIEW

File: tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/context/BMainInstance.java [hunk 1]
```diff
@@ -413,12 +441,7 @@ private boolean isLeechingDone(ServerLogReader infoReader, ServerLogReader error
      * @param cmdArgs ballerina command args.
      */
     private void terminateProcess(Process process, String[] cmdArgs) {
```
Comment (NipunaRanasinghe) [inline tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/context/BMainInstance.java:443]: `terminateProcess(Process process, String[] cmdArgs)` method can be further simplified using its overloaded method as follows.

```java
private void terminateProcess(Process process, String[] cmdArgs) {
        // Extracts the debug port from the command arguments.
        String port = "";
        for (int i = 0; i < cmdArgs.length; i++) {
            if (cmdArgs[i].equals("--debug")) {
                port = cmdArgs[i + 1];
                terminateProcess(process, port);
                break;
            }
        }
    }
```

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/build/MultiModuleBuildDebugTest.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    @Test
+    public void testSuspendOnBallerinaModuleBuild() throws BallerinaTestException {
+        port = findFreePort();
+        String msg = "Listening for transport dt_socket at address: " + port;
+        LogLeecher clientLeecher = new LogLeecher(msg);
+        process = balClient.debugMain("build", new String[]{"--debug", String.valueOf(port), testModuleName},
+                null, new String[]{}, new LogLeecher[]{clientLeecher}, projectPath, 20,
+                true);
+        clientLeecher.waitForText(20000);
+    }
```
Comment (NipunaRanasinghe) [inline tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/build/MultiModuleBuildDebugTest.java:68]: AFAIR, we have already covered this test scenario in remote debug tests suite. Shall we remove this?

File: tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/context/BMainInstance.java (hunk 1 above)
Comment (praveennadarajah) [reply tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/context/BMainInstance.java:443]: Fixed

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/build/MultiModuleBuildDebugTest.java (hunk 2 above)
Comment (praveennadarajah) [reply tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/build/MultiModuleBuildDebugTest.java]: Fixed

**Pull request merged**